src/main/java/CalculatorTesting/InfixToPostfixTests.java

```diff
@@ -0,0 +1,46 @@
+package CalculatorTesting;
+
+import ShuntingYardCalculator.Calculator.CalculationSteps.InfixToPostfix;
+import ShuntingYardCalculator.Enums.Type;
+import javafx.util.Pair;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+
+import static CalculatorTesting.TestingParsers.ParsePostfixString.parse;
+import static ShuntingYardCalculator.Calculator.EquationParser.parseEquationString;
+import static ShuntingYardCalculator.ExceptionType.EMPTY_BRACKETS_ERROR;
+
+public class InfixToPostfixTests {
+    @Test
+    public void complexDoubleBracketsChallenge() {
+        ArrayList<Pair<String, Type>> complexDoubleInput = parseEquationString("5+((5+5)*5)");
+        ArrayList<Pair<String, Type>> complexDoubleExpected = parse("5 5 5 + 5 * +");
+
+        ArrayList<Pair<String, Type>> complexDoubleOutput = InfixToPostfix.infixToPostfix(complexDoubleInput);
+
+        Assert.assertEquals(complexDoubleExpected, complexDoubleOutput);
+    }
+
+    @Test
+    public void doubleEmptyBracketsWarning() {
+        ArrayList<Pair<String, Type>> doubleEmptyInput = parseEquationString("(())");
+
+        try {
+            InfixToPostfix.infixToPostfix(doubleEmptyInput);
+        } catch (ArithmeticException e) {
```

Comment:
https://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests